tests/raftstore/test_multi.rs

```diff
@@ -121,6 +125,49 @@ fn test_multi_lost_majority<T: Simulator>(cluster: &mut Cluster<T>, count: usize
 
 }
 
+fn test_multi_random_restart<T: Simulator>(cluster: &mut Cluster<T>,
+                                           node_count: usize,
+                                           restart_count: u32) {
+    cluster.bootstrap_region().expect("");
+    cluster.start();
+
+    let (key, value) = (b"a1", b"v1");
+
+    assert_eq!(cluster.get(key), None);
+    cluster.put(key, value);
+    assert_eq!(cluster.get(key), Some(value.to_vec()));
+
+    let mut rng = rand::thread_rng();
+    for _ in 1..restart_count {
+        let id = 1 + rng.gen_range(0, node_count as u64);
+        cluster.stop_node(id);
+        cluster.run_node(id);
+        wait_until_node_online(cluster, id);
+        assert_eq!(cluster.get(key), Some(value.to_vec()));
+
+        // verify whether data is actually being replicated
+        must_get_equal(&cluster.get_engine(id), key, value);
+    }
+}
+
+fn wait_until_node_online<T: Simulator>(cluster: &mut Cluster<T>, node_id: u64) {
```

Comment:
@siddontang 
we must send find_leader command to ensure **raft leader is elected out**, which is what `wait_until_node_online` does. while `must_get_equal ` check engine DB storage directly, it knows nothing about raft cluster.

if we remove this function,  just use `must_get_equal` to ensure node online and put or get after that, we'll get "{error {message: "peer is not leader" not_leader {region_id: 1}}"